### PR TITLE
FIX: Support hashing of FILES

### PR DESCRIPTION
**Pull request by djape007**
Properties used for hashing FILE are:
name, size, type and lastModified

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -223,35 +223,39 @@ function typeHasher(options, writeTo, context){
...
-        // seem to break because of that.
-        if (options.respectType !== false && !isNativeFunction(object)) {
-          keys.splice(0, 0, 'prototype', '__proto__', 'constructor');
-        }
-
-        if (options.excludeKeys) {
-          keys = keys.filter(function(key) { return !options.excludeKeys(key); });
-        }
+        if (objType === 'file') {
+          write('file:' + object.name + "," + object.size + "," + object.type + "," + object.lastModfied);
```
Comment (addaleax) [inline index.js:227]: I think hashing the content would be the most important part here

File: index.js [hunk 2]
```diff
@@ -214,7 +214,7 @@ function typeHasher(options, writeTo, context){
         return write(object);
       }
 
-      if(objType !== 'object' && objType !== 'function' && objType !== 'asyncfunction') {
+      if(objType !== 'object' && objType !== 'function' && objType !== 'asyncfunction' && objType !== 'file') {
         if(this['_' + objType]) {
           this['_' + objType](object);
```
Comment (addaleax) [inline index.js:219]: Is there any particular reason for side-stepping the regular dispatch mechanism using the underscore-prefixed methods in this file?

Comment (djape007) [reply index.js:219]: Not really, I just wasn't payng attention.

File: index.js (hunk 1 above)
Comment (djape007) [reply index.js:227]: I agree. How to access file's content?

Comment (addaleax) [reply index.js:227]: Hm, I’m not sure that that’s possible to do synchronously (file.arrayBuffer() is `async`) – so probably this is the best we can do

**Review by addaleax — APPROVED** [review]

**Comment by addaleax** [comment]
Are you able to add a test, or should somebody else do that?

**Comment by djape007** [comment]
Sorry, i can't do it in any reasonable time frame.

File: index.js [hunk 3]
```diff
@@ -392,6 +392,10 @@ function typeHasher(options, writeTo, context){
       var arr = Array.from(set);
       return this._array(arr, options.unorderedSets !== false);
     },
+    _file: function(file) {
+      write('file:');
+      return this.dispatch([file.name, file.size, file.type, file.lastModfied]);
```
Comment (18alantom) [inline index.js:397]: Typo found: it should be `file.lastModified` not `file.lastModfied`

**Pull request merged**